salt/orch/update.sls

```diff
@@ -294,9 +336,10 @@ all-masters-post-start-services:
   salt.state:
     - tgt: '{{ worker_id }}'
     - sls:
+      - migrations.2-3.kubelet.stop
```

Comment:
I miss some more granularity in the versions numbers for migrations. I think it would be more future-proof to have something like `migrations.2.0-3.0.kubelet.stop`.

By doing that, maybe we could do in the future some more flexible migrations with something like:

```SaltStack
{%- set worker_id_osrelease = salt.caasp_grains.get(worker_id, 'osrelease')[worker_id] %}

{{ worker_id }}-early-clean-shutdown:
   salt.state:
     - tgt: '{{ worker_id }}'
     - sls:
      - migrations.{{ worker_id_osrelease }}-3.0.kubelet.stop
   ...
```

that would choose the right migration depending on the current CaaSP version.

Reply:
Despite I like the idea we cannot do like that (as the dot `.` is the path separator) -- your example would go to look into `migrations/2/0-3/0/kubelet/stop.sls`.

I think the best we could achieve would be something like `migrations.2-0-3-0.kubelet.stop`, but it's a bit weird, what do you think?

Reply:
Oh my... what @inercia proposed was really nice, too bad we cannot do it :man_facepalming: 

I don't know what feels less weird between the two proposals @ereslibre made, IMO I would go with `migrations.2-0-3-0.kubelet.stop`

Reply:
Yes, with `major-minor` scheme we could transform the `osrelease` grain replacing dots by dashes and we could have the same feature.

I will update the PR.

Reply:
We can create a `get()` in `_module/caas_osrelease.py` with something like

```Python
from caasp_grains import get as get_grain

# the separator for versions, like 2_1 insstead of 2.1
VERSION_SEP = '_'

def get(host):
    return str(get_grain(host, 'osrelease')[host]).replace('.'. VERSION_SEP)
```

so we could just do:

```SaltStack
{{ worker_id }}-early-clean-shutdown:
   salt.state:
     - tgt: '{{ worker_id }}'
     - sls:
      - migrations.{{ salt.caasp_osrelease.get(worker_id) }}-3_0.kubelet.stop
```

then we should have a `migrations.2_1-3_0.kubelet.stop` file

Reply:
I wrote some thoughts about migrations here: https://github.com/kubic-project/salt/pull/626#discussion_r200917311